core_v2/sources/domains.move

```diff
@@ -0,0 +1,1232 @@
+    const COLLECTION_URI: vector<u8> = b"https://aptosnames.com";
+    /// current MAX_REMAINING_TIME_FOR_RENEWAL_SEC is 6 months
+    const MAX_REMAINING_TIME_FOR_RENEWAL_SEC: u64 = 15552000;
+    /// 2024/03/07 23:59:59
+    const SECONDS_PER_YEAR: u64 = 60 * 60 * 24 * 365;
+
+    /// enums for subdomain expiration policy. update validate_subdomain_expiration_policy() when adding more
+    const SUBDOMAIN_POLICY_MANUAL_SET_EXPIRATION: u8 = 0;
+    const SUBDOMAIN_POLICY_LOOKUP_DOMAIN_EXPIRATION: u8 = 1;
+    // const SUBDOMAIN_POLICY_NEXT_ENUM = 2
+
+    /// The Naming Service contract is not enabled
+    const ENOT_ENABLED: u64 = 1;
+    /// The caller is not authorized to perform this operation
+    const ENOT_AUTHORIZED: u64 = 2;
+    /// The name is not available, as it has already been registered
+    const ENAME_NOT_AVAILABLE: u64 = 3;
+    /// The number of years the caller attempted to register the domain or subdomain for is invalid
+    const EINVALID_NUMBER_YEARS: u64 = 4;
+    /// The domain does not exist- it is not registered
+    const ENAME_NOT_EXIST: u64 = 5;
+    /// The caller is not the owner of the domain, and is not authorized to perform the action
+    const ENOT_OWNER_OF_DOMAIN: u64 = 6;
+    /// The caller is not the owner of the name, and is not authorized to perform the action
+    const ENOT_OWNER_OF_NAME: u64 = 9;
+    /// The domain name is too long- it exceeds the configured maximum number of utf8 glyphs
+    const ENAME_TOO_LONG: u64 = 10;
+    /// The domain is too short.
+    const ENAME_TOO_SHORT: u64 = 11;
+    /// The domain name contains invalid characters: it is not a valid domain name
+    const ENAME_HAS_INVALID_CHARACTERS: u64 = 12;
+    /// The required `register_domain_signature` for `register_domain` is missing
+    const EVALID_SIGNATURE_REQUIRED: u64 = 16;
+    /// The name is not expired in 6 months, thus not eligible for renewal
+    const EDOMAIN_NOT_AVAILABLE_TO_RENEW: u64 = 18;
+    /// The subdomain is not eligible for renewal
+    const ESUBDOMAIN_IS_AUTO_RENEW: u64 = 19;
+    /// The name is expired
+    const ENAME_EXPIRED: u64 = 20;
+    /// The subdomain not exist
+    const ESUBDOMAIN_NOT_EXIST: u64 = 21;
+    /// The name is not a subdomain
+    const ENOT_A_SUBDOMAIN: u64 = 22;
+    /// The subdomain expiration can be set any date before the domain expiration
+    const ESUBDOMAIN_EXPIRATION_PASS_DOMAIN_EXPIRATION: u64 = 24;
+    /// The duration must be whole years
+    const EDURATION_MUST_BE_WHOLE_YEARS: u64 = 25;
+    /// The subdomain expiration policy is included in the enum SUBDOMAIN_POLICY_*
+    const ESUBDOMAIN_EXPIRATION_POLICY_INVALID: u64 = 26;
+    /// Caller must be the router
+    const ENOT_ROUTER: u64 = 27;
+
+    /// Tokens require a signer to create, so this is the signer for the collection
+    struct CollectionCapability has key, drop {
+        capability: SignerCapability,
+        burn_signer_capability: SignerCapability,
+    }
```

Comment:
Do we really want to continue to leverage this legacy method for managing the service?